plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelViewImpl.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.ext.git.client.panel;
+
+import static com.google.common.collect.Iterables.getFirst;
+
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwt.user.client.ui.Widget;
+import java.util.Comparator;
+import java.util.List;
+import javax.inject.Inject;
+import org.eclipse.che.ide.api.parts.base.BaseView;
+import org.eclipse.che.ide.ext.git.client.GitLocalizationConstant;
+import org.eclipse.che.ide.ext.git.client.GitResources;
+import org.eclipse.che.ide.ext.git.client.compare.changespanel.ChangesPanelView;
+import org.eclipse.che.ide.ext.git.client.panel.GitPanelView.ActionDelegate;
+import org.eclipse.che.ide.ui.smartTree.NodeLoader;
+import org.eclipse.che.ide.ui.smartTree.NodeStorage;
+import org.eclipse.che.ide.ui.smartTree.NodeStorage.StoreSortInfo;
+import org.eclipse.che.ide.ui.smartTree.SelectionModel;
+import org.eclipse.che.ide.ui.smartTree.SortDir;
+import org.eclipse.che.ide.ui.smartTree.Tree;
+import org.eclipse.che.ide.ui.smartTree.data.Node;
+
+/** @author Mykola Morhun */
+public class GitPanelViewImpl extends BaseView<ActionDelegate> implements GitPanelView {
+  interface GitPanelViewImplUiBinder extends UiBinder<Widget, GitPanelViewImpl> {}
+
+  @UiField SimplePanel changesPanel;
+
+  @UiField(provided = true)
+  final GitResources gitResources;
+
+  @UiField(provided = true)
+  final GitLocalizationConstant locale;
+
+  @UiField(provided = true)
+  Tree repositoriesList;
+
+  private final RepositoryNodeFactory repositoryNodeFactory;
+
+  @Inject
+  public GitPanelViewImpl(
+      GitPanelViewImplUiBinder uiBinder,
+      RepositoryNodeFactory repositoryNodeFactory,
+      GitResources gitResources,
+      GitLocalizationConstant locale) {
+    this.repositoryNodeFactory = repositoryNodeFactory;
+    this.gitResources = gitResources;
+    this.locale = locale;
+
+    createRepositoriesList();
+    setContentWidget(uiBinder.createAndBindUi(this));
+  }
+
+  private void createRepositoriesList() {
```

Comment:
The general approach: private methods are located after the public ones

Reply:
Generally - yes. This method here because it is part of constructor, just moved to separate method for clarification.

Reply:
general approach would be the rather the stepdown rule (as the human would read your code) http://tidyjava.com/stepdown-rule/